dependencies/jcifs/pom.xml

```diff
@@ -16,10 +16,15 @@
   </description>
   <dependencies>
     <dependency>
-      <groupId>org.samba.jcifs</groupId>
-      <artifactId>jcifs</artifactId>
+      <groupId>eu.agno3.jcifs</groupId>
+      <artifactId>jcifs-ng</artifactId>
       <version>${jcifsVersion}</version>
     </dependency>
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcprov-jdk15on</artifactId>
+      <version>${bcprovVersion}</version>
```

Comment:
Here be dragons: https://issues.opennms.org/browse/NMS-7959

As mentioned in JIRA, we've seen problems with Bouncycastle before, but that may of just been related to an outdated version.

Reply:
Is there any record of the cipher suites in use in case that resulted in that issue so we could test it? Also it seems like [priorities of providers can be managed](https://docs.oracle.com/javase/8/docs/technotes/guides/security/crypto/CryptoSpec.html#Security) in a JVM, although I'm not sure how that works in the land of karaf etc.

Reply:
The previous problems we encountered were against bcprov-jdk14-1.38.jar - from BouncyCastle 2.28.1 - which was released on November 7 2007. The issue was reported in November 2015 - so we were likely just running against an outdated version.

Given this is being targeted for a .0 in a major release and I'd be comfortable pulling it back into the default class-path.

Reply:
Sounds good. 

I did a bit more reading, and they do note in the [docs](https://www.bouncycastle.org/documentation.html) that the unrestricted JCE policy files are required in order for it to function properly. Those policy files have been provided with the Oracle JDK since 8u161.